src/format/opamTypes.mli

```diff
@@ -273,21 +284,14 @@ type user_action =
 
 (** Solver universe *)
 type universe = {
+  u_pefuniv  : ((string * string), Opam.Packages.package) Hashtbl.t;
+  u_options  : (string * string list * string list);
+  u_tables   : Pef.Pefcudf.tables;
```

Comment:
No problem then, currently the conversion makes it quite costly to convert from opam-state to cudf universe, which is annoying e.g. when you want to just compute a set of transitive dependencies or a topological sort. We already use Graph.imperative -- with a functional interface on top.

It should just be documented, and maybe the types named: what are all the strings ? I am curious about the `u_options` too.

So using hashtables is not a problem, but it's good to know whether they can be muted or not: it has implications on the validity of a computed universe.